numba/compiler.py

```diff
@@ -460,21 +462,31 @@ def _backend(self, lowerfn, objectmode):
         if self.library is None:
             codegen = self.targetctx.jit_codegen()
             self.library = codegen.create_library(self.bc.func_qualname)
-        lowered = lowerfn()
+        if config.DISABLE_JIT and not self.flags.force_enable_jit:
```

Comment:
If we just want to disable compilation when functions are decorated with `@jit` or `@njit`, then it would not be necessary to disable it here. I added this here in case there are some users who use `compile_isolated` instead of the decorators.